backend/app/utils.py

```
@@ -1852,3 +2684,719 @@ def _process_user_event(portal_instance, event_id, operation):
+    if (directory / "Index.json").exists():
+        return True
+
+    # Check a sample XML file to see if it contains JSON
+    xml_files = list(directory.rglob("*.xml"))[:3]  # Check first 3 XML files
+    for xml_file in xml_files:
+        try:
+            with open(xml_file, 'r', encoding='utf-8') as f:
+                content = f.read(100).strip()
+                if content.startswith('{') or content.startswith('['):
+                    return True
+        except Exception:
```

Ruff (0.14.10)

Do not catch blind exception: `Exception`

(BLE001)